### ISSUE TITLE
Rule for enforcing use of passive voice

### ISSUE THREAD

**Issue by DerekNonGeneric** [nodejs/remark-preset-lint-node#289]
In professional documentation, I expect the passive voice. It’s not clear who the “we” would be here.

_Originally posted by @GeoffreyBooth in https://github.com/nodejs/node/pull/40708#discussion_r748957366_

---

Strong agree with @GeoffreyBooth from me &mdash; perhaps we should consider including [`retext-passive`](https://www.npmjs.com/package/retext-passive) in our preset to assist us in catching these.

**Comment by Trott** [comment]
I'm going to try to be understated on this: Oh, hell no.

First of all, and least importantly, `retext-passive` does the opposite of what you are asking for here. It flags passive voice as a problem.

Second, the recommendation for active-voice "we recommend" over passive-voice "it is recommended" is from the Microsoft Style Guide. The reason we adopted the Microsoft Style Guide was to avoid this kind of style-guide second-guessing. Sure, we need to make sure people spell it _Node.js_ and not _NodeJS_. That needs to be in our local style guide. But again, oh, hell no to trying to work out within the group whether it's better to use serial commas or not, whether passive voice or active voice is preferred, and whether "may" is preferable to "might" or the other way around.

The Microsoft Style Guide, in an effort to make documentation friendly/approachable/informal, [recommends active voice for most content](https://docs.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice).   Let's not second-guess them.

By the way, when this decision was being made, the reason I proposed the Microsoft Style Guide was because [that's what MDN uses](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide#preferred_style_guides), and MDN seemed to me to be the gold standard for online JavaScript documentation. We link to it all over the docs when we need to link to types and things like that.

In a nutshell, I'm sorry if you find the style too informal or otherwise not what you would choose, but at the same time, please, let's not do this.

**Comment by Trott** [comment]
Oh, by the way, if you want to try to get your change in upstream, you can open an issue or a pull request at https://github.com/MicrosoftDocs/microsoft-style-guide. If you can make the case for a change or (more likely) an exception/clarification that would apply to the case that started this conversation, then those changes would naturally be adopted by Node.js over time.

**Comment by GeoffreyBooth** [comment]
I’m surprised to learn that the Microsoft style guide discourages the passive voice. I thought it was pretty standard that computers say things like “File not found” and not “I could not find the file” or “Windows could not find the file” or whatever. I thought the same extended to documentation.

Anyway this is why we follow a larger standard, to avoid debates like this. This isn’t a Node-specific thing, so if it’s a settled question then we shouldn’t make an exception for ourselves.

**Comment by Trott** [comment]
> I thought it was pretty standard that computers say things like “File not found” and not “I could not find the file” or “Windows could not find the file” or whatever.

They explicitly call out error messages such as these as a situation where passive voice is appropriate. See https://docs.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice where the first bullet under uses for passive voice says that it is useful "especially in errors, warnings, or notifications".

**Comment by Trott** [comment]
I'm going to close this, but please feel free to re-open or comment if the issue is not in fact settled. Thanks!

**Comment by DerekNonGeneric** [comment]
> First of all, and least importantly, `retext-passive` does the opposite of what you are asking for
> here. It flags passive voice as a problem.

Maybe the plugin fooled me since this lint rule seems to successfully enforce a similar-but-different stylistic problem that requires eliminating informal pronouns from academic and professional writing (likely the actual underlying issue both myself and Geoffrey are interested in resolving). I found a couple of good examples in a pamphlet issued by academia:

- In our opinion, the price of gasoline is too high.
- We expect all work to be turned in on time.

It is possible to write opinions or give directions without personal pronouns [“our” & “we”)]:

- The price of gasoline is too high.
- All work is to be turned in on time.

Refs: https://www.ius.edu/writing-center/files/finding-proper-pronouns.pdf

> They explicitly call out error messages like that as a situation where passive voice is appropriate. > See https://docs.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice where the first bullet under uses for passive voice says that it is useful “especially in errors, warnings, or notifications”.

Thanks for sharing the link again. I’m still not familiar w/ our current style guide (the MS Docs one) since I lost interest after learning that they don’t even use title-case for page titles. 😉

OTOH, that exception you mention sounds pretty sensible and may even be something worth enforcing.

Ensuring the use of passive voice at least for the messages that appear in developer’s consoles and on our [“Errors” API reference docs page](https://nodejs.org/api/errors.html) could be facilitated via a remark lint plugin enforcing the opposite of what `remark-passive` is expected to achieve &mdash; something that ensures the use of a de-personalized active voice.

> I’m going to close this, but please feel free to re-open or comment if the issue is not, in fact, settled. Thanks!

I would be happy to re-open this issue (perhaps after first renaming it to include the added specificity) in hopes we can apply such a rule to assist w/ verifying error message compliance (it seems like necessary privileges in this repo need to be restored).

**Comment by DerekNonGeneric** [comment]
> First of all, and least importantly, `retext-passive` does the opposite of what you are asking for
> here. It flags passive voice as a problem.

Maybe the plugin fooled me since this lint rule seems to successfully enforce a similar-but-different stylistic problem that requires eliminating informal pronouns from academic and professional writing (likely the actual underlying issue both myself and Geoffrey are interested in resolving). I found a couple of good examples in a pamphlet issued by academia:

- In our opinion, the price of gasoline is too high.
- We expect all work to be turned in on time.

It is possible to write opinions or give directions without personal pronouns [the “our” & “we”]:

- The price of gasoline is too high.
- All work is to be turned in on time.

Refs: https://www.ius.edu/writing-center/files/finding-proper-pronouns.pdf

> They explicitly call out error messages like that as a situation where passive voice is appropriate.
> See https://docs.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice where the first bullet under uses for passive voice says that it is useful “especially in errors, warnings, or notifications”.

Thanks for sharing the link again. I’m still not familiar w/ our current style guide (the MS Docs one) since I lost interest after learning that they don’t even use title-case for page titles. 😉

OTOH, that exception you mention sounds pretty sensible and may even be something worth enforcing.

Ensuring the use of passive voice at least for the messages that appear in developers’ consoles and on our [“Errors” API reference docs page](https://nodejs.org/api/errors.html) could be facilitated via a remark lint plugin enforcing the opposite of what `remark-passive` is expected to achieve &mdash; something that ensures the use of a de-personalized active voice.

> I’m going to close this, but please feel free to re-open or comment if the issue is not, in fact, settled. Thanks!

I would be happy to re-open this issue (perhaps after first renaming it to include the added specificity) in hopes we can apply such a rule to assist w/ verifying error message compliance (it seems like necessary privileges in this repo need to be restored).
